src/scripts/quests/QuestLineHelper.ts

```diff
@@ -2078,6 +2078,118 @@ class QuestLineHelper {
 
 
     // Alola QuestLines
+    // Melemele Island guide - Started upon deafting Hau 1
+    public static createMelemeleAlolaQuestLine() {
+        const melemeleAlolaQuestLine = new QuestLine('Welcome to paradise, cousin!', 'See the sights this small tropical island has to offer!');
+        // 0 - Talk to Lillie at Mahalo Trail (Melemele Woods)
+        const talkToLillie1 = new TalkToNPCQuest(Lillie1, 'Time to kick off your Alolan vacation! They say the sacred Mahalo Trail in Melemele Woods behind Iki Town is home to Melemele\'s legendary guardian. Perhaps a fateful encounter awaits?');
+        melemeleAlolaQuestLine.addQuest(talkToLillie1);
+
+        // 1 - Fight Spearow
+        const battleMelemeleSpearow = new CustomQuest (1, 0,  'Protect the mysterious girl\'s Pokémon! Battle the Spearow on Mahalo Trail.', () => App.game.statistics.temporaryBattleDefeated[GameConstants.getTemporaryBattlesIndex('Melemele Spearow')](), undefined, undefined,
+            {
+                clearedMessage: '<i>The bridge collapses beneath you and the floating Pokémon, and you both fall to the turbulent cascades below... But a swift creature zaps to your rescue! It seems fixated on the safety of the little one.</i>',
+                npcDisplayName: 'Melemele Guardian',
+                npcImageName: '../pokemon/785',
+            });
+        melemeleAlolaQuestLine.addQuest(battleMelemeleSpearow);
+
+        // 2 - Return Nebby to Lillie
+        const talkToLillie2 = new TalkToNPCQuest(Lillie2, 'Almost falling to your doom was not on the brochure. This never would have happened in Kanto. Ask the mysterious girl in white if it\'s too late for a refund.');
+        melemeleAlolaQuestLine.addQuest(talkToLillie2);
+
+        // 3 - Talk to Recon Squad in Iki
+        const talkToReconSquad1 = new TalkToNPCQuest(ReconSquad1, 'Talk to tourists.');
+        const talkToReconSquad2 = new TalkToNPCQuest(ReconSquad2, 'Talk to more tourists.');
+        melemeleAlolaQuestLine.addQuest(new MultipleQuestsQuest([
+            talkToReconSquad1,
+            talkToReconSquad2,
+        ],'Now that you\'re all acquainted, time to enjoy Iki Town\'s festival! Mingle with some fellow tourists after battling Hau.'));
```

Comment:
You can talk with some of the tourists before beating hau. Not sure if that is intended or not?